Vagrantfile

```diff
@@ -9,63 +9,40 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # options are documented and commented below. For a complete reference,
   # please see the online documentation at vagrantup.com.
 
+  #calculate memory allocation
+
+  hostmem = Integer(%x[ grep MemTotal /proc/meminfo ].scan(/\d+/).shift) / 1024
+  if hostmem >= 8000 then 
+    dev_mem = 4096
+  elsif hostmem < 4000 and hostmem > 2048 then 
+    dev_mem = 2048
+  else
+    dev_mem = 1024 
+  end
```

Comment:
One thought more... is better this levels (1GB, 2GB and 4GB) or just divide by 2 the hostmem with a maximum of 4GB??